storage/aptosdb/src/db_debugger/truncate/mod.rs

```diff
@@ -103,14 +118,18 @@ impl Cmd {
             println!("Done!");
         }
 
-        println!("Starting ledger db truncation...");
-        let ledger_db = Arc::new(ledger_db);
-        truncate_ledger_db(
-            Arc::clone(&ledger_db),
-            ledger_db_version,
-            self.target_version,
-            self.ledger_db_batch_size,
+        println!("Starting ledger db and state kv db truncation...");
+        let batch = SchemaBatch::new();
+        batch.put::<DbMetadataSchema>(
+            &DbMetadataKey::OverallCommitProgress,
+            &DbMetadataValue::Version(self.target_version),
         )?;
+        ledger_db.write_schemas(batch)?;
```

Comment:
looks like there's no drawback to use `DB::put()` directly as we discussed :P

Reply:
Done.